pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,2086 @@
+        return command.format_map({self.placeholder: ord(self.id) - 64})
+
+    def clear_measurement_buffer(self):
+        """ Clears the measurement data buffer (``MBC``). """
+        self.write('mbc {{ch}}')
+
+    def set_output_type(self, output_type, measurement_type):
+        """ Sets the output method and type of the GPIB output (``OFM``).
+
+        :param output_type: A property that controls the type of output
+        :type output_type: int or :class:`OutputType`
+        :param measurement_type: A property that controls the measurement type
+        :type measurement_type: int or :class:`MeasurementType`
+
+        .. note::
+
+            For the format of the output data, refer to :meth:`AdvantestR624X.set_output_format`.
+            For DC and pulse measurements, the output method is fixed to '1' (real-time output).
+            When the output method '3' (buffering output) is specified, the measured data is not
+            stored in memory.
+
+        """
+        output_type = OutputType(output_type)
+        measurement_type = MeasurementType(measurement_type)
+        self.write(f'ofm {{ch}},{output_type.value},{measurement_type.value}')
+
+    analog_input = Channel.setting(
+        "fl {ch},%d",
+        """ Set the analog input terminal (ANALOG INPUT) on the rear panel ON or OFF (``FL``).
+
+        :type: int
+
+            1. Turn off the analog input.
+            2. Analog input ON, gain x1.
+            3. Analog input ON, gain x2.5.
+
+        """,
+        validator=strict_range,
+        values=range(1, 3),
+    )
+
+    trigger_output_timing = Channel.setting(
+        "tot {ch},%d",
+        """ Set the timing of the trigger output signal
+        output from TRIGGER OUT on the rear panel (``TOT``).
+        the status in the form of a :class:`TriggerOutputSignalTiming` ``IntFlag``.
+
+        :type: :class:`.TriggerOutputSignalTiming`
+
+        """,
+        validator=strict_range,
+        values=range(0, 63),
+        get_process=lambda v: TriggerOutputSignalTiming(int(v)),
+    )
+
+    def set_scanner_control(self, output, interlock):
+        """  Sets the SCANNER CONTROL (START, STOP)
+        output signal and INTERLOCK input signal on the rear panel (``SCT``).
+
+        :param int output: A property that controls the scanner output
+        :param int interlock: A property that controls the scanner interlock type
+
+        output:
+
+            1. Scanner, Turn off the control signal output.
+            2. Output to the scanner control signal at the start / stop of the sweep.
+            3. Operate / Standby Scanner, Output to the control signal.
+
+        interlock:
+
+            1. Turn off the interlock signal input.
+            2. Set as a stamper when the interlock signal input is HI.
+            3. When the interlock signal input is HI, it is on standby, and when it is LO, it is
+               operated.
+
+        """
+        output = truncated_range(output, [1, 3])
```

Comment:
I would use strict discrete set, as that raises an error instead of truncating. And it will also check, that you don't give a float like 1.5.